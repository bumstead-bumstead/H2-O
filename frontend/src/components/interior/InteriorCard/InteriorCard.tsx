import type { ComponentProps } from 'react';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { Card, Flex, Icon, Typography } from '@/components/common';
import { toPriceFormatString } from '@/utils/string';

interface Props extends ComponentProps<typeof Card> {
  choiceRatio: number;
  name: string;
  price: number;
  image: string;
}

function InteriorCard({ isSelected, choiceRatio, name, price, image, ...restProps }: Props) {
  return (
    <StyledCard isSelected={isSelected} {...restProps}>
      <img src={image} alt={name} width='69px' height='100%' />
      <Content flexDirection='column'>
        <Typography font='TextKRRegular12' color='gray600'>
          <Typography as='span' font='TextKRMedium12' color={isSelected ? 'activeBlue' : 'gray600'}>
            {choiceRatio}%
          </Typography>
          가 선택했어요
        </Typography>
        <Typography as='h3' font='HeadKRMedium14' color={isSelected ? 'gray900' : 'gray500'}>
          {name}
        </Typography>
        <Typography font='TextKRMedium14' color={isSelected ? 'gray900' : 'gray600'} marginTop={24}>
          {toPriceFormatString(price)} 원
        </Typography>
        <Icon
          iconType='Check'
          color={isSelected ? 'activeBlue' : 'gray200'}
          css={css`
            position: absolute;
            bottom: 8px;
            right: 24px;
          `}
        />
      </Content>
    </StyledCard>
  );
}

export default InteriorCard;

const StyledCard = styled(Card)`
  display: flex;
  width: 244.5px;
  height: 110px;
  border-radius: 2px;
`;

const Content = styled(Flex)`
  position: relative;
  flex: 1;
  height: 100%;
  padding: 14px 0px 8px 16px;
  background-color: transparent;
`;
