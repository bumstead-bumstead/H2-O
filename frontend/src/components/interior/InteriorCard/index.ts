export { default as InteriorCard } from './InteriorCard';
