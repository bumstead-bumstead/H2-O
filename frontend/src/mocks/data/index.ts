export * from './trims';
export * from './technical-spec';
export * from './selectionInfo';
export * from './priceRange';
export * from './model-type';
export * from './interiors';
