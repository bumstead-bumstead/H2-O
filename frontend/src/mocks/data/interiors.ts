export const interiors = [
  {
    id: 1,
    name: '퀼팅 천연(블랙)',
    choiceRatio: 62,
    price: 0,
    fabricImage: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/colorchip-interior-black.png',
    bannerImage: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/img-interior-black.png',
  },
  {
    id: 2,
    name: '쿨그레이',
    choiceRatio: 38,
    price: 0,
    fabricImage: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/colorchip-interior-gray.png',
    bannerImage: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/img-interior-gray.png',
  },
];
