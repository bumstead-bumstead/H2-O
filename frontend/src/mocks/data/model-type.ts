export const modelType = {
  powertrains: [
    {
      id: 1,
      name: '디젤 2.2',
      price: 1480000,
      choiceRatio: 48,
      description: '높은 토크로 파워풀한 드라이빙이 가능하며, 차급대비 연비 효율이 우수합니다',
      image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/powertrain/dieselengine2.2.jpg',
      maxOutput: {
        output: 202,
        minRpm: 3800,
        maxRpm: 3800,
      },
      maxTorque: {
        torque: 45,
        minRpm: 1750,
        maxRpm: 2750,
      },
    },
    {
      id: 2,
      name: '가솔린 3.8',
      price: 0,
      choiceRatio: 62,
      description:
        '고마력의 우수한 가속 성능을 확보하여, 넉넉하고 안정감 있는 주행이 가능합니다\n엔진의 진동이 적어 편안하고 조용한 드라이빙 감성을 제공합니다',
      image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/powertrain/gasoline3.8.jpg',
      maxOutput: {
        output: 295,
        minRpm: 6000,
        maxRpm: 6000,
      },
      maxTorque: {
        torque: 36.2,
        minRpm: 5200,
        maxRpm: 5200,
      },
    },
  ],
  bodytypes: [
    {
      id: 1,
      name: '7인승',
      description:
        '기존 8인승 시트(1열 2명, 2열 3명, 3열 3명)에서 2열 가운데 시트를 없애 2열 탑승객의 편의는 물론, 3열 탑승객의 승하차가 편리합니다',
      image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/bodytype/7_seat.jpg',
      price: 0,
      choiceRatio: 45,
    },
    {
      id: 2,
      name: '8인승',
      description: '1열 2명, 2열 3명, 3열 3명이 탑승할 수 있는 구조로, 많은 인원이 탑승할 수 있도록 배려하였습니다',
      image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/bodytype/8_seat.jpg',
      price: 0,
      choiceRatio: 55,
    },
  ],
  drivetrains: [
    {
      id: 1,
      name: '2WD',
      description:
        '엔진에서 전달되는 동력이 전/후륜 바퀴 중 한쪽으로만 전달되어 차량을 움직이는 방식입니다\n차체가 가벼워 연료 효율이 높습니다',
      image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/drivetrain/2wd.png',
      price: 0,
      choiceRatio: 22,
    },
    {
      id: 2,
      name: '4WD',
      description:
        '엔진에서 전달되는 동력이 전/후륜 바퀴 중 한쪽으로만 전달되어 차량을 움직이는 방식입니다\n차체가 가벼워 연료 효율이 높습니다',
      image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/drivetrain/4wd.png',
      price: 2370000,
      choiceRatio: 78,
    },
  ],
};
