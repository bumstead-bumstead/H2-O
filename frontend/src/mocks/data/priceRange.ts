export const priceRange = { maxPrice: 54590000, minPrice: 38960000 };
