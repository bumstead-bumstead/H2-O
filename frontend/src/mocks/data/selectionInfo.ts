import { type State } from '@/providers/SelectionProvider/hooks/useSelectionReducer';

export const selectionInfo: State = {
  model: {
    id: 1,
    name: '팰리세이드',
    price: 0,
  },
  trim: {
    id: 2,
    name: 'Le Blanc',
    price: 38960000,
  },
  priceRange: { maxPrice: 54590000, minPrice: 38960000 },
  powerTrain: {
    id: 1,
    image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/powertrain/dieselengine2.2.jpg',
    name: '디젤 2.2',
    price: 1480000,
  },
  bodyType: {
    id: 1,
    image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/bodytype/7_seat.jpg',
    name: '7인승',
    price: 0,
  },
  driveTrain: {
    id: 1,
    image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/drivetrain/2wd.png',
    name: '2WD',
    price: 0,
  },
  exteriorColor: {
    colorCode: '#121929',
    id: 3,
    image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/trim/le-blanc/blue_exterior/image_001.png',
    name: '문라이트 블루 펄',
    price: 0,
  },
  interiorColor: {
    fabricImage: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/colorchip-interior-black.png',
    id: 1,
    image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/img-interior-black.png',
    name: '퀼팅 천연(블랙)',
    price: 0,
  },
  displacement: 2199,
  fuelEfficiency: 12.16,
  extraOptions: {
    optionList: [
      {
        id: 1,
        image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/package/roa.jpg',
        isPackage: true,
        isQuotation: false,
        name: '컴포트 Ⅱ',
        price: 1090000,
      },
      {
        id: 68,
        image: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/option/builtincam.jpg',
        isPackage: false,
        isQuotation: false,
        name: '빌트인 캠(보조배터리 포함)',
        price: 690000,
      },
    ],
    price: 1780000,
  },
};

export const calculateTotalPrice = (selectionInfo: State) => {
  const { trim, powerTrain, bodyType, driveTrain, exteriorColor, interiorColor, extraOptions } = selectionInfo;
  const options = [trim, powerTrain, bodyType, driveTrain, exteriorColor, interiorColor, extraOptions];

  return options.reduce((acc, curr) => acc + (curr?.price || 0), 0);
};
