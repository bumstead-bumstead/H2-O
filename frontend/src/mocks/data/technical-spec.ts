export const technicalSpecs = {
  displacement: 3778,
  fuelEfficiency: 9.23,
};
