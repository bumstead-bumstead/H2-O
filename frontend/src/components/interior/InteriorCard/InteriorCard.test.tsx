import InteriorCard from './InteriorCard';
import { interiors } from '@/mocks/data';
import { render, screen } from '@/tests/test-util';

describe('InteriorCard 컴포넌트 테스트', () => {
  const { name, price, choiceRatio, fabricImage: image } = interiors[0];
  const props = {
    choiceRatio,
    name,
    price,
    image,
  };

  it('InteriorCard 컴포넌트가 정상적으로 렌더링 된다.', () => {
    render(<InteriorCard data-testid='card' {...props} />);

    expect(screen.getByTestId('card')).toBeInTheDocument();
  });

  it('props에 따라 카드에 나타나는 텍스트가 달라집니다.', () => {
    render(<InteriorCard data-testid='card' {...props} />);
    const card = screen.getByTestId('card');

    expect(card).toHaveTextContent(`${choiceRatio}%가 선택했어요`);
    expect(card).toHaveTextContent(`${name}`);
    expect(card).toHaveTextContent(`+0 원`);
  });

  describe('InteriorCard 스냅샷 테스트', () => {
    it('isSelected가 false일 때', () => {
      render(<InteriorCard data-testid='card' {...props} />);

      expect(screen.getByTestId('card')).toMatchSnapshot();
    });

    it('isSelected가 true일 때', () => {
      render(<InteriorCard data-testid='card' isSelected {...props} />);

      expect(screen.getByTestId('card')).toMatchSnapshot();
    });
  });
});
