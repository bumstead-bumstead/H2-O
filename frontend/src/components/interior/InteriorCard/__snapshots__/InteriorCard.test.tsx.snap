// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InteriorCard 컴포넌트 테스트 > InteriorCard 스냅샷 테스트 > isSelected가 false일 때 1`] = `
.emotion-0 {
  border: 1px solid #DADCDD;
  background-color: white;
  color: #A3A6AA;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 244.5px;
  height: 110px;
  border-radius: 2px;
}

.emotion-0:hover {
  border-color: #00AAD2;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  padding: 14px 0px 8px 16px;
  background-color: transparent;
}

.emotion-2 {
  color: #838588;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
  letter-spacing: -0.03em;
}

.emotion-3 {
  color: #838588;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 12px;
  font-weight: 500;
  line-height: 18px;
  letter-spacing: -0.03em;
}

.emotion-4 {
  color: #A3A6AA;
  font-family: Hyundai Sans Head KR,sans-serif;
  font-size: 14px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: -0.03em;
}

.emotion-5 {
  color: #838588;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 14px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: -0.03em;
  margin-top: 24px;
}

.emotion-6 {
  position: absolute;
  bottom: 8px;
  right: 24px;
}

<div
  class="emotion-0"
  data-testid="card"
>
  <img
    alt="퀼팅 천연(블랙)"
    height="100%"
    src="https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/colorchip-interior-black.png"
    width="69px"
  />
  <div
    class="emotion-1"
  >
    <p
      class="emotion-2"
      color="gray600"
    >
      <span
        class="emotion-3"
        color="gray600"
      >
        62
        %
      </span>
      가 선택했어요
    </p>
    <h3
      class="emotion-4"
      color="gray500"
    >
      퀼팅 천연(블랙)
    </h3>
    <p
      class="emotion-5"
      color="gray600"
    >
      +0
       원
    </p>
    <svg
      class="emotion-6"
      data-testid="Check"
      fill="#DADCDD"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m9.7 18.025-5.7-5.7L5.425 10.9 9.7 15.175 18.875 6 20.3 7.425l-10.6 10.6Z"
        fill="inherit"
      />
    </svg>
  </div>
</div>
`;

exports[`InteriorCard 컴포넌트 테스트 > InteriorCard 스냅샷 테스트 > isSelected가 true일 때 1`] = `
.emotion-0 {
  border: 1px solid #00AAD2;
  background-color: #00AAD21A;
  color: #212122;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 244.5px;
  height: 110px;
  border-radius: 2px;
}

.emotion-0:hover {
  border-color: #00AAD2;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  padding: 14px 0px 8px 16px;
  background-color: transparent;
}

.emotion-2 {
  color: #838588;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
  letter-spacing: -0.03em;
}

.emotion-3 {
  color: #00AAD2;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 12px;
  font-weight: 500;
  line-height: 18px;
  letter-spacing: -0.03em;
}

.emotion-4 {
  color: #212122;
  font-family: Hyundai Sans Head KR,sans-serif;
  font-size: 14px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: -0.03em;
}

.emotion-5 {
  color: #212122;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 14px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: -0.03em;
  margin-top: 24px;
}

.emotion-6 {
  position: absolute;
  bottom: 8px;
  right: 24px;
}

<div
  class="emotion-0"
  data-testid="card"
>
  <img
    alt="퀼팅 천연(블랙)"
    height="100%"
    src="https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/colorchip-interior-black.png"
    width="69px"
  />
  <div
    class="emotion-1"
  >
    <p
      class="emotion-2"
      color="gray600"
    >
      <span
        class="emotion-3"
        color="activeBlue"
      >
        62
        %
      </span>
      가 선택했어요
    </p>
    <h3
      class="emotion-4"
      color="gray900"
    >
      퀼팅 천연(블랙)
    </h3>
    <p
      class="emotion-5"
      color="gray900"
    >
      +0
       원
    </p>
    <svg
      class="emotion-6"
      data-testid="Check"
      fill="#00AAD2"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m9.7 18.025-5.7-5.7L5.425 10.9 9.7 15.175 18.875 6 20.3 7.425l-10.6 10.6Z"
        fill="inherit"
      />
    </svg>
  </div>
</div>
`;
